Duplicati/Library/Backend/Sia/Sia.cs

```diff
@@ -83,6 +83,11 @@ private System.Net.HttpWebRequest CreateRequest(string endpoint)
 
         private string getResponseBodyOnError(string context, System.Net.WebException wex)
         {
+            if (wex.Response is null)
```

Comment:
Should this be `if (wex.Response as System.Net.HttpWebResponse == null)` ?
Then it would also catch cases where the response is a netwtork error, rather than crash on the casting.

Maybe consider rewriting it as:
```csharp
var response = wex.Response as System.Net.HttpWebResponse;
if (response == null)
  return $"{context} failed with error: {wex.Message}";

```

Reply:
Yes, good idea.  Revision 0d991d571266a322f01898c133a771ebd290b8d4 refines the check.